packages/nx/src/project-graph/build-dependencies/explicit-project-dependencies.ts

```diff
@@ -34,7 +36,10 @@ export function buildExplicitTypeScriptDependencies(
               // TODO: These edges technically should be allowed but we need to figure out how to separate config files out from root
               return;
             }
-
+            // standalone apps should not have dependency to jest.preset.js imports
+            if (isRoot(source) && IGNORED_ROOT_FILES.includes(f.file)) {
```

Comment:
Why is this an issue?

Reply:
Currently generated `package.json` for standalone `fastify` app looks like this:

```json
{
  "name": "fastify-demo",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "start": "nx serve",
    "build": "nx build",
    "test": "nx test"
  },
  "private": true,
  "dependencies": {
    "axios": "^1.0.0",
    "tslib": "^2.3.0"
  },
  "devDependencies": {
    "@nrwl/esbuild": "15.7.0",
    "@nrwl/eslint-plugin-nx": "15.7.0",
    "@nrwl/jest": "15.7.0",
    "@nrwl/linter": "15.7.0",
    "@nrwl/node": "15.7.0",
    "@nrwl/nx-cloud": "latest",
    "@nrwl/workspace": "15.7.0",
    "@types/jest": "28.1.1",
    "@types/node": "18.7.1",
    "@typescript-eslint/eslint-plugin": "^5.36.1",
    "@typescript-eslint/parser": "^5.36.1",
    "esbuild": "^0.17.5",
    "eslint": "~8.15.0",
    "eslint-config-prettier": "8.1.0",
    "fastify": "4.11.0",
    "jest": "28.1.1",
    "jest-environment-jsdom": "28.1.1",
    "nx": "15.7.0",
    "prettier": "^2.6.2",
    "ts-jest": "28.0.5",
    "ts-node": "10.9.1",
    "typescript": "~4.8.2"
  },
  "main": "./main.cjs"
}
```

With explicit check for root `project.json` we end up with this:
```jsonc
{
  "name": "fastify-demo",
  "version": "0.0.0",
  "dependencies": {
    "@nrwl/jest": "15.7.0", // this comes from `jest.preset.js`
    "fastify": "4.11.0"
  },
  "main": "./main.cjs"
}
```

What we want to achieve is:
```json
{
  "name": "fastify-demo",
  "version": "0.0.0",
  "dependencies": {
    "fastify": "4.11.0"
  },
  "main": "./main.cjs"
}
```

This will not be an issue once we filter out test files and their configuration, but at the moment, it's adding unnecessary dependencies.